### PR TITLE
Allow ignore workbook corruption error

### PR DESCRIPTION
**Pull request by skonik**
Solves the issues
https://github.com/python-excel/xlrd/issues/285
https://github.com/python-excel/xlrd/issues/149

My first contribution ever.

### CODE REVIEW

File: tests/test_ignore_workbook_corruption_error.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    def test_not_corrupted(self):
+        with self.assertRaises(Exception) as context:
+            xlrd.open_workbook(from_this_dir('corrupted_error.xls'))
+        self.assertTrue('Workbook corruption' in str(context.exception))
+
+        try:
+            xlrd.open_workbook(from_this_dir('corrupted_error.xls'), ignore_workbook_corruption_error=True)
+        except CompDocError:
+            self.fail('ignore_workbook_corruption_error=True, but the exception is raised.')
```
Comment (cjw296) [inline tests/test_ignore_workbook_corruption_error.py]: I wouldn't bother with the try/except here, the original exception will be more useful.

File: xlrd/__init__.py [hunk 2]
```diff
@@ -38,7 +38,9 @@ def open_workbook(filename=None,
                   encoding_override=None,
                   formatting_info=False,
                   on_demand=False,
-                  ragged_rows=False):
+                  ragged_rows=False,
+                  ignore_workbook_corruption_error=False
```
Comment (cjw296) [inline xlrd/__init__.py]: I think it'd be better to shorten this to just `ignore_workbook_corruption`.

File: tests/test_ignore_workbook_corruption_error.py (hunk 1 above)
Comment (skonik) [reply tests/test_ignore_workbook_corruption_error.py]: Did you mean just leave `xlrd.open_workbook(from_this_dir('corrupted_error.xls'), ignore_workbook_corruption_error=True)`?

Comment (skonik) [reply tests/test_ignore_workbook_corruption_error.py]: > I wouldn't bother with the try/except here, the original exception will be more useful.

Did you mean just leave` xlrd.open_workbook(from_this_dir('corrupted_error.xls'), ignore_workbook_corruption_error=True)`?

**Comment by benco5** [comment]
> Solves the issues https://github.com/python-excel/xlrd/issues/285 https://github.com/python-excel/xlrd/issues/149
> 
> My first contribution ever.

Is there a way I can see these issues? When reading `.xls` with pandas, xlrd throws a corruption error that I believe may be false. Having access to past issues helps.

**Comment by skonik** [comment]
> @cjw296
> 
> > Solves the issues https://github.com/python-excel/xlrd/issues/285 https://github.com/python-excel/xlrd/issues/149
> > My first contribution ever.
> 
> Is there a way I can see these issues? When reading `.xls` with pandas, xlrd sometimes throws a corruption error that I believe may be false. Having access to past issues helps.

Not sure what happened to the repo and how you can access these issues.

What's your goal? As far as I remember a few people were reading some excel files and got corruption error, however they were able to read them in other software.

Can suggest to change engine if possible. 

Screenshot 2023-04-23 at 23 47 57

Otherwise somehow to pass additional kwargs `ignore_workbook_corruption=True` to `xlrd` engine.

**Comment by benco5** [comment]
Hi @skonik Yes, well aware of the ability to specify the engine. I was curious whether the linked issues I'm unable to view were similar / same as mine and others that I've been seeing people ask about. As a sidenote, very recent PR in pandas (unsure if yet released) allows `ignore_workbook_corruption=True` to be passed, I believe as `engine_options`. Anyway, regardless, I was playing around with `xlrd` directly and passing `ignore_workbook_corruption=True`, which does prevent the 'corruption' error, but the rows end up truncated, I believe unnecessarily (i.e., `xlrd` inability to handle certain `.xls` format).  I was able to trackdown a way to read my file(s) to completion and without error, so this further leads me to think that `xlrd` can be improved to accommodate such situations, but unfortunately it seems I cannot view nor open an issue. Anyway, thanks for your reply!
